uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/actionmenu/ActionMenuBottomDialogFragment.kt

```diff
@@ -0,0 +1,169 @@
+
+    private var viewModel: ActionMenuInternalViewModel? = null
+
+    private lateinit var binding: LayoutActionMenuInternalBinding
+
+    /**
+     * Set a [ActionMenuData] for the [ActionMenuBottomDialogFragment] content fulfillment.
+     *
+     * @param [ActionMenuData] which will be used for fulfillment the [ActionMenuBottomDialogFragment] content.
+     */
+    var data: ActionMenuData
+        get() = arguments?.getParcelable(ACTION_MENU_DATA) ?: ActionMenuData.empty
+        set(value) {
+            arguments?.putParcelable(ACTION_MENU_DATA, value)
+            viewModel?.actionMenuData = value
+        }
+
+    /**
+     * Register a callback to be invoked when a [ActionMenuBottomDialogFragment] menu item click has been made.
+     *
+     * @param [ActionMenuItemClickListener] callback to invoke [ActionMenuBottomDialogFragment] item click.
+     */
+    var itemClickListener: ActionMenuItemClickListener? = null
+        get() = viewModel?.listener ?: field
+        set(value) {
+            field = value
+            viewModel?.let { it.listener = value }
+        }
+
+    /**
+     * A *[spanCount]* define the number of columns to be used. The default value is [DEFAULT_SPAN_COUNT].
+     *
+     * @param [Int] wanted number of columns.
+     *
+     * @return [Int] current columns value.
+     */
+    var spanCount: Int
+        get() = arguments?.getInt(ACTION_MENU_SPAN_COUNT) ?: DEFAULT_SPAN_COUNT
+        set(value) {
+            arguments?.putInt(ACTION_MENU_SPAN_COUNT, value)
+            viewModel?.spanCount!!.value = value
+        }
+
+    /**
+     * @see ActionMenuBottomDialogFragment
+     */
+    companion object {
+
+        const val TAG = "action_menu_bottom_dialog_fragment"
+
+        /**
+         * Allows you to simply create new instance of [ActionMenuBottomDialogFragment]. You can provide a [ActionMenuData]
+         * object and spanCount value.
+         *
+         * @param actionMenuData [ActionMenuData] to be applied to the dialog content.
+         * @param spanCount [Int] to be applied to the dialog content.
+         */
+        @JvmStatic
+        fun newInstance(
+            actionMenuData: ActionMenuData,
+            spanCount: Int = DEFAULT_SPAN_COUNT
+        ): ActionMenuBottomDialogFragment = ActionMenuBottomDialogFragment().apply {
+            arguments = Bundle().apply {
+                putInt(ACTION_MENU_SPAN_COUNT, spanCount)
+                putParcelable(ACTION_MENU_DATA, actionMenuData)
+            }
+        }
+    }
+
+    override fun getDialog(): BottomSheetDialog = super.getDialog() as BottomSheetDialog
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewModel =
+            ViewModelProviders.of(this@ActionMenuBottomDialogFragment).get(ActionMenuInternalViewModel::class.java)
+                .apply {
+                    this.spanCount.value = arguments.getInt(ACTION_MENU_SPAN_COUNT, DEFAULT_SPAN_COUNT)
```

Comment:
nemalo by to byt skor cez factory ako si spominal?

Reply:
nakonec jsem tu neudělal factory, protože všechny ty věci jsou konfigurovatelné i později a nic specifického ten VM momentálně nevyžaduje.